modules/cuckooml/cuckooml.py

```diff
@@ -0,0 +1,1677 @@
+        self.simple_features , self.simple_features_description = \
+            self.extract_simple_features(simple_features)
+
+
+    def export_simple_dataset(self, filename="simple_dataset.csv"):
+        """Export a dataset consisting of malware labels and *simple* features
+        to CSV formatted file."""
+        # Check if data and labels are loaded
+        if self.simple_features is None:
+            print "Please load simple features first."
+            return
+
+        if self.labels is None:
+            print "Please load labels first."
+            return
+
+        simple_dataset = pd.concat([self.simple_features, self.labels], axis=1)
+        simple_dataset.to_csv(filename)
+
+
+    def simple_feature_category(self, category="properties"):
+        """Get simple feature data frame containing only features form selected
+        category."""
+        if self.simple_features is None:
+            print "Simple features are not loaded. Please load them before \
+                   extracting categories."
+            return None
+
+        return self.simple_features.loc[:, self.SIMPLE_CATEGORIES[category]]
+
+
+    def extract_features(self, features, include_API_calls=False, \
+                      include_API_calls_count=False):
+        """Extract features form an external object into pandas data frame."""
+        my_features = {}
+        for i in features:
+            my_features[i] = {}
+
+            # Exponentially bin the binary size and timestamp
+            my_features[i][":meta:size"] = self.__log_bin(features[i]["size"])
+            my_features[i][":meta:timestamp"] = \
+                self.__log_bin(features[i]["timestamp"])
+
+            # Handle ExifTool output
+            for j in self.__handle_string(features[i]["FileDescription"]):
+                my_features[i][":meta:" + j] = 1
+            for j in self.__handle_string(features[i]["OriginalFilename"]):
+                my_features[i][":meta:" + j] = 1
+            for j in self.__handle_string(features[i]["magic_byte"]):
+                my_features[i][":meta:" + j] = 1
+
+            # Is the binary signed?
+            my_features[i][":sign:signed"] = int(features[i]["signed"])
+            # And other signature features
+            for j in self.__handle_string(features[i]["Comments"]):
+                my_features[i][":sign:" + j] = 1
+            for j in self.__handle_string(features[i]["ProductName"]):
+                my_features[i][":sign:" + j] = 1
+            for j in self.__handle_string(features[i]["LegalCopyright"]):
+                my_features[i][":sign:" + j] = 1
+            for j in self.__handle_string(features[i]["InternalName"]):
+                my_features[i][":sign:" + j] = 1
+            for j in self.__handle_string(features[i]["CompanyName"]):
+                my_features[i][":sign:" + j] = 1
+
+            # Extract packer
+            patterns = [r"Armadillo", r"PECompact", r"ASPack", r"ASProtect",
```

Comment:
If you could move this to a class variable or somewhere else outside the actual function, that'd improve the code a bit :)